testsuite/MDAnalysisTests/coordinates/test_imd.py

```diff
@@ -0,0 +1,418 @@
+
+    def test_volume(self, ref, reader):
+        # don't rewind here as in inherited base test
+        vol = reader.ts.volume
+        # Here we can only be sure about the numbers upto the decimal point due
+        # to limited floating point precision.
+        assert_allclose(vol, ref.volume, rtol=0, atol=1.5e0)
+
+    def test_reload_auxiliaries_from_description(self, ref, reader):
+        pytest.skip("Cannot create two IMDReaders on the same stream")
+
+    def test_stop_iter(self, reader):
+        pytest.skip("IMDReader cannot be rewound")
+
+    def test_iter_rewinds(self, reader):
+        pytest.skip("IMDReader cannot be rewound")
+
+    def test_timeseries_shape(self, reader):
+        pytest.skip("IMDReader does not support timeseries")
+
+    def test_timeseries_asel_shape(self, reader):
+        pytest.skip("IMDReader does not support timeseries")
+
+    def test_timeseries_values(self, reader):
+        pytest.skip("IMDReader does not support timeseries")
+
+    def test_transformations_2iter(self, ref, transformed):
+        pytest.skip("IMDReader cannot be reopened")
+
+    def test_transformations_slice(self, ref, transformed):
+        pytest.skip("IMDReader cannot be reopened")
+
+    def test_transformations_switch_frame(self, ref, transformed):
+        pytest.skip("IMDReader cannot be reopened")
+
+    def test_transformation_rewind(self, ref, transformed):
+        pytest.skip("IMDReader cannot be reopened")
+
+    def test_pickle_reader(self, reader):
+        pytest.skip("IMDReader cannot be pickled")
+
+    def test_pickle_next_ts_reader(self, reader):
+        pytest.skip("IMDReader cannot be pickled")
+
+    def test_pickle_last_ts_reader(self, reader):
+        pytest.skip("IMDReader cannot be pickled")
+
+    def test_transformations_copy(self, ref, transformed):
+        pytest.skip("IMDReader cannot be copied")
+
+    def test_timeseries_empty_asel(self, reader):
+        pytest.skip("IMDReader does not support timeseries")
+
+    def test_timeseries_empty_atomgroup(self, reader):
+        pytest.skip("IMDReader does not support timeseries")
+
+    def test_timeseries_asel_warns_deprecation(self, reader):
+        pytest.skip("IMDReader does not support timeseries")
+
+    def test_timeseries_atomgroup(self, reader):
+        pytest.skip("IMDReader does not support timeseries")
+
+    def test_timeseries_atomgroup_asel_mutex(self, reader):
+        pytest.skip("IMDReader does not support timeseries")
+
+    def test_last_frame(self, ref, reader):
+        pytest.skip("IMDReader cannot be rewound")
+
+    def test_go_over_last_frame(self, ref, reader):
+        pytest.skip("IMDReader must be an indexed using a slice")
+
+    def test_frame_jump(self, ref, reader):
+        pytest.skip("IMDReader must be an indexed using a slice")
+
+    def test_frame_jump_issue1942(self, ref, reader):
+        pytest.skip("IMDReader must be an indexed using a slice")
```

Comment:
add tests that check for the correct exceptions raised for the examples of incorrect slices in https://mdanalysis--4923.org.readthedocs.build/en/4923/documentation_pages/coordinates/base.html#MDAnalysis.coordinates.base.StreamReaderBase.check_slice_indices namely

```
traj[5:]       # Cannot specify start index
traj[:100]     # Cannot specify stop index
traj[5:100:2]  # Cannot specify start or stop indices
traj[::-1]     # Cannot go backwards (negative step)
```

I'd create a parametrized test for that.

Reply:
Ok – tested below in `test_iterate_start_stop_raises_error` and others.